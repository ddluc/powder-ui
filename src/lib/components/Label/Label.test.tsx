import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Label, LabelProps } from './index';
import withTheme from '../../../../test/withTheme';

// Define unit tests
describe('Label', () => {
  afterEach(cleanup);

  describe('as a static label', () => {
    const args: LabelProps = {
      value: 'First Name',
      htmlFor: 'first-name',
      animated: false
    };
    it('should render', () => {
      const result = withTheme(<Label {...args} />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
    it('should render with disabled state', () => {
      const result = withTheme(<Label {...args} disabled />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
    it('should render with active state', () => {
      const result = withTheme(<Label {...args} isActive />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
    it('should render with error state', () => {
      const result = withTheme(<Label {...args} error />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
  });

  describe('as an animated label', () => {
    const args: LabelProps = {
      value: 'First Name',
      htmlFor: 'first-name',
      animated: true,
      position: { x: 1, y: 2 },
      transpose: { x: 0, y: -20 }
    };
    it('should render', () => {
      const result = withTheme(<Label {...args} />);
      const component = result.container.firstChild;
      expect(component).toMatchSnapshot();
    });
  });
});
