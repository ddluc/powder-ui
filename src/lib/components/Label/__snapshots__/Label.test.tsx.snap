// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label as a static label should render 1`] = `
.c1 {
  font-size: 12px;
  display: auto;
  color: #3E4242;
}

.c0 {
  display: block;
  position: relative;
  text-align: left;
}

<label
  class="c0"
  for="first-name"
>
  <span
    class="c1"
  >
    First Name
  </span>
</label>
`;

exports[`Label as a static label should render with active state 1`] = `
.c1 {
  font-size: 12px;
  display: auto;
  color: #3E4242;
}

.c0 {
  display: block;
  position: relative;
  text-align: left;
}

<label
  class="c0"
  for="first-name"
>
  <span
    class="c1"
  >
    First Name
  </span>
</label>
`;

exports[`Label as a static label should render with disabled state 1`] = `
.c1 {
  font-size: 12px;
  display: auto;
  color: #C4C4C4;
}

.c0 {
  display: block;
  position: relative;
  text-align: left;
}

<label
  class="c0"
  for="first-name"
>
  <span
    class="c1"
    disabled=""
  >
    First Name
  </span>
</label>
`;

exports[`Label as a static label should render with error state 1`] = `
.c1 {
  font-size: 12px;
  display: auto;
  color: #DB7B71;
}

.c0 {
  display: block;
  position: relative;
  text-align: left;
}

<label
  class="c0"
  for="first-name"
>
  <span
    class="c1"
  >
    First Name
  </span>
</label>
`;

exports[`Label as an animated label should render 1`] = `
.c1 {
  position: absolute;
  display: initial;
  top: 2px;
  left: 1px;
  font-size: 14px;
  max-width: 100%;
  padding: 0px 2px;
  margin: 0px;
  -webkit-transition: color .1s cubic-bezier(0,0,0.2,1), -webkit-transform .1s cubic-bezier(0,0,0.2,1);
  -webkit-transition: color .1s cubic-bezier(0,0,0.2,1), transform .1s cubic-bezier(0,0,0.2,1);
  transition: color .1s cubic-bezier(0,0,0.2,1), transform .1s cubic-bezier(0,0,0.2,1);
  z-index: 6;
  background: #FFFFFF;
  color: #A5B0AE;
}

.c0 {
  display: block;
  position: relative;
  text-align: left;
}

<label
  class="c0"
  for="first-name"
>
  <span
    class="c1"
  >
    First Name
  </span>
</label>
`;
